sklearn/naive_bayes.py

```diff
@@ -951,3 +952,101 @@ def _joint_log_likelihood(self, X):
         jll += self.class_log_prior_ + neg_prob.sum(axis=1)
 
         return jll
+
+
+class CategoricalNB(BaseDiscreteNB):
+
+    def __init__(self, alpha=1.0, fit_prior=True, class_prior=None):
+        self.alpha = alpha
+        self.fit_prior = fit_prior
+        self.class_prior = class_prior
+
+    # is overridden because we use a different implementation for self.feature_count
+    def fit(self, X, y, sample_weight=None):
+        """Fit Naive Bayes classifier according to X, y
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
```

Comment:
You definitely need to document the type of X here.

Reply:
Alright. I did it in commit 'fix tests for py 2.7'. Unfortunately, I forgot to commit the doc string separately.